deno/v8/payloads/interactions.ts

```diff
@@ -115,17 +115,24 @@ export interface APIApplicationCommandInteractionData {
 /**
  * https://discord.com/developers/docs/interactions/slash-commands#interaction-applicationcommandinteractiondataoption
  */
-export interface APIApplicationCommandInteractionDataOption {
-	name: string;
-	/**
-	 * The value returned here depends on the `ApplicationCommandOptionType` type of the option with the name
-	 * that matches this interface's `name`.
-	 *
-	 * You will need to manually cast this to the appropriate type based on the returned data
-	 */
-	value?: unknown;
-	options?: APIApplicationCommandInteractionDataOption[];
-}
+export type APIApplicationCommandInteractionDataOption =
+	| { name: string; options: APIApplicationCommandInteractionDataOption[] }
+	| {
+			name: string;
+			type: ApplicationCommandOptionType.SUB_COMMAND | ApplicationCommandOptionType.SUB_COMMAND_GROUP;
+			options: APIApplicationCommandInteractionDataOption[];
+	  }
+	| {
+			name: string;
+			type:
+				| ApplicationCommandOptionType.STRING
+				| ApplicationCommandOptionType.USER
+				| ApplicationCommandOptionType.CHANNEL
+				| ApplicationCommandOptionType.ROLE;
+			value: string;
+	  }
```

Comment:
separate into one for `string`s and one for `Snowflake`s

Reply:
I didn't realize there was as Snowflake type, I'll do that.